com.unity.sg2/Tests/HeadlessPreview/AllNodesTest.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using NUnit.Framework;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor.ShaderGraph.GraphDelta;
+using System.Linq;
+using UnityEditor.ShaderGraph.HeadlessPreview.UnitTests;
+using UnityEngine.TestTools;
+using static UnityEditor.VersionControl.Message;
+
+namespace UnityEditor.ShaderGraph.HeadlessPreview.NodeTests
+{
+
+    // TODO: Consider async compile/rendering, need to wait until thread is done and results are returned
+    // https://forum.unity.com/threads/support-for-async-await-in-tests.787853/#post-6838493
+
+    [TestFixture]
+    class DefaultPreviewsTestFixture
+    {
+        static ShaderGraphRegistry InitSGR()
+        {
+            ShaderGraphRegistry SGR = new();
+            SGR.InitializeDefaults();
+            return SGR;
+        }
+
+        static string[] InitNodeNames()
+        {
+            return SGR.DefaultTopologies.GetNodes().Select(e => e.ID.LocalPath).ToArray();
+        }
+
+        // whether or not a node will require another node to coerce the value to a vector for testing preview compilation.
+        static bool HasVectorOutput(NodeHandler node)
+        {
+            var nodeKey = node.GetRegistryKey();
+            var typeField = node.GetPorts().Where(e => !e.IsInput)?.First()?.GetTypeField();
+            bool hasPreviewableOutputPort = typeField?.GetRegistryKey().Name == GraphType.kRegistryKey.Name;
+            return hasPreviewableOutputPort && GraphTypeHelpers.GetHeight(typeField) == GraphType.Height.One;
+        }
+
+        static ShaderGraphRegistry SGR = InitSGR();
```

Comment:
This is the only time `InitSGR` is called. Can you combine them?

Reply:
What do you mean combine them? This is mostly just to make sure the SGR can be initialized statically.

Reply:
Merging this with my other comment (see other comment).